### PR TITLE
Early admin portal

### PR DESCRIPTION
**Pull request by hey-waffles**

### CODE REVIEW

File: src/migrations/indexes/rulesets_by_official.fauna [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Creates an index for listing either officially supported rulesets or unofficial rulesets
+CreateIndex({
+  name: "rulesets_by_official",
+  source: Collection("rulesets"),
+  terms: [{ field: ["data", "isOfficial"]} ],
+  values: [
+    { field: ["ref"], reverse: true},
```
Comment (hey-waffles) [inline src/migrations/indexes/rulesets_by_official.fauna:7]: Should fetch by updatedAt to give us the most recent changes

File: src/pages/admin/rulesets/[id].tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import { InitialProps } from "types/client";
+import { RulesetDocument } from "types/documents";
+import { getSession } from "utilities/auth";
+import { rest } from "utilities/request";
+
+interface AdminRulesetProps extends InitialProps {
+  ruleset: RulesetDocument;
+}
+
+function AdminRuleset(props: AdminRulesetProps) {
```
Comment (hey-waffles) [inline src/pages/admin/rulesets/[id].tsx:20]: Comment

File: src/pages/admin/rulesets/[id].tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  const [ ruleset, setRuleset ] = React.useState(props.ruleset);
+
+  React.useEffect(() => {
```
Comment (hey-waffles) [inline src/pages/admin/rulesets/[id].tsx:23]: Comment

File: src/pages/admin/rulesets/[id].tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    RulesetManager.load();
+    RulesetManager.set(props.ruleset);
+  }, []);
+
+  React.useEffect(() => {
```
Comment (hey-waffles) [inline src/pages/admin/rulesets/[id].tsx:28]: Comment

File: src/pages/admin/rulesets/index.tsx [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+interface AdminRulesetsProps extends InitialProps {
+  rulesets: RulesetDocument[];
+}
+
+
+interface RulesetUpdateResponse {
+  ruleset: RulesetDocument;
+}
+
+async function setPublic(id: string, isPublic: boolean) {
```
Comment (hey-waffles) [inline src/pages/admin/rulesets/index.tsx:32]: Comment

File: src/pages/admin/rulesets/index.tsx [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+
+  RulesetManager.set(result.data.ruleset);
+  return;
+}
+
+interface RulesetOwnerProps {
+  ownedBy?: UserDocument;
+}
+
+function RulesetOwner(props: RulesetOwnerProps) {
```
Comment (hey-waffles) [inline src/pages/admin/rulesets/index.tsx:47]: Comment

File: src/server/auth.ts [hunk 7]
```diff
@@ -6,9 +6,10 @@ import { toFaunaRef } from "utilities/fauna";
 export function getMyUser(req: NextApiRequest): MyUserDocument {
   // TODO - have this working on page refresh
   const myUser = getSession({req});
-
+  console.log(myUser)
   // If someone is not logged in
   if (myUser === null || !("user" in myUser)) {
+    console.log("WHHYYY")
```
Comment (hey-waffles) [inline src/server/auth.ts:12]: Remove

File: src/server/auth.ts [hunk 8]
```diff
@@ -6,9 +6,10 @@ import { toFaunaRef } from "utilities/fauna";
 export function getMyUser(req: NextApiRequest): MyUserDocument {
   // TODO - have this working on page refresh
   const myUser = getSession({req});
-
+  console.log(myUser)
```
Comment (hey-waffles) [inline src/server/auth.ts:9]: Remove

File: src/server/faunaLogicBuilder/helpers/index.ts [hunk 9]
```diff
@@ -40,6 +40,8 @@ export function checkConfig(config: unknown) {
  */
 export function canActStatic(myUser: MyUserDocument, roleConfig: RoleConfig) {
   // TODO - Check if role is not present in myUser?
+  console.log(myUser.role)
+  console.log(roleConfig[myUser.role])
```
Comment (hey-waffles) [inline src/server/faunaLogicBuilder/helpers/index.ts:44]: Remove

File: src/server/faunaLogicBuilder/helpers/index.ts [hunk 10]
```diff
@@ -53,6 +55,7 @@ export function canActStatic(myUser: MyUserDocument, roleConfig: RoleConfig) {
  */
 export function canActOn(docs: AnyDocument[], myUser: MyUserDocument, roleConfig: RoleConfig): AnyDocument[] {
   const approvedDocs: AnyDocument[] = [];
+  console.log(myUser.roles)
```
Comment (hey-waffles) [inline src/server/faunaLogicBuilder/helpers/index.ts:58]: Remove

File: src/server/faunaLogicBuilder/methods/search.ts [hunk 11]
```diff
@@ -46,9 +46,12 @@ export async function $search(
     throw { code: 500, message: `An error occured while trying to search the ${config.index} index` };
   }
   let docs = parseIndexResponse(result.data, config.indexFields);
+  console.log("post parse")
```
Comment (hey-waffles) [inline src/server/faunaLogicBuilder/methods/search.ts:49]: remove

File: src/server/faunaLogicBuilder/methods/search.ts [hunk 12]
```diff
@@ -46,9 +46,12 @@ export async function $search(
...
+  console.log(docs)
   // Returns a subset of docs that the current user can see
   docs = canActOn(docs, myUser, config.roles);
-
+  console.log("post can act on")
+  console.log(docs)
```
Comment (hey-waffles) [inline src/server/faunaLogicBuilder/methods/search.ts:54]: remove

File: src/server/faunaLogicBuilder/methods/update.ts [hunk 13]
```diff
@@ -23,7 +23,7 @@ export async function $update(
 ) {
   const client = getServerClient();
   const newRef = toFaunaRef(ref, config.collection);
-
+  console.log(doc)
```
Comment (hey-waffles) [inline src/server/faunaLogicBuilder/methods/update.ts:26]: remove

File: src/server/faunaLogicBuilder/methods/update.ts [hunk 14]
```diff
@@ -44,8 +44,8 @@ export async function $update(
   // Preprocesses, setting certain fields and doing any required steps before updating
   doc = config.preProcess(doc, myUser);
   doc = preUpdate(doc, myUser, config) as unknown as AnyDocument;
-
-  let result: FaunaDocument<unknown> | null = await client.query(q.Update(faunaDoc.ref as Expr, doc));
+  console.log(doc)
```
Comment (hey-waffles) [inline src/server/faunaLogicBuilder/methods/update.ts:47]: Remove. Why did I use so many console logs???

File: src/server/faunaLogicBuilder/methods/update.ts [hunk 15]
```diff
@@ -58,7 +58,7 @@ export async function $update(
   result = config.postProcess(result, myUser);
   const fields = determineAccessibleFields(result as AnyDocument, myUser, config.fields[getRole(myUser)]);
   result = trimRestrictedFields(result as Record<string, unknown>, fields, true);
-
+  console.log(result)
```
Comment (hey-waffles) [inline src/server/faunaLogicBuilder/methods/update.ts:61]: remove

File: src/server/faunaLogicBuilder/methods/update.ts [hunk 16]
```diff
@@ -69,6 +69,7 @@ export async function $update(
  * @returns An updated document ready to be updated
  */
 function preUpdate(doc: AnyDocument, myUser: MyUserDocument, config: FunctionConfig) {
+  console.log(doc)
```
Comment (hey-waffles) [inline src/server/faunaLogicBuilder/methods/update.ts:72]: Remove

File: src/migrations/indexes/rulesets_by_official.fauna (hunk 1 above)
Comment (hey-waffles) [reply src/migrations/indexes/rulesets_by_official.fauna:7]: Resolved, but the Data Manager needs a way of indexing and sorting by different fields

**Review by lucyawrey — APPROVED** [review]
Looks good

**Pull request merged**